data/patches/miq_application.js.diff

```diff
@@ -35,5 +35,60 @@
 +// --------------
 +
 +
++// setTimeout and clearTimeout monitoring
++// ---------------
++
++// Set up container to hold timeout and interval state
++window.testing = {
++  timeouts: {}
++};
++
++// Make a copy of the original setTimeout function
++window._setTimeout = window.setTimeout;
++window.setTimeout = function(callback, timeout) {
++  // We need a handle to store our timeout under, we can't just use the
++  // timeout ID because we don't know it until after we create the timeout
++  var handle = _.uniqueId();
++
++  // Call the old setTimeout function
++  var timeoutId = window._setTimeout(
++    function() {
++      // The callback is the function we were originally deferring
++      callback();
++
++      // Once a timeout completes, we need to remove our reference to it
++      delete window.testing.timeouts[handle];
++    },
++    timeout
++  );
++
++  // Store the id of the timeout we just created so it can be queried
++  window.testing.timeouts[handle] = timeoutId;
```

Comment:
`window._setTimeout` schedules an event, returns id of the timer and we store that id. So we know the id to store only after we call `window._setTimeout`. The actual callback passed to window._setTimeout is called in the future (after `timeout`).